robot/src/main/java/us/ilite/robot/driverinput/DriverInput.java

```diff
@@ -61,87 +68,108 @@ public void periodicInput(double pNow) {
-                trackingType = ETrackingType.HATCH_LEFT;
+    private void updateDriveTrain() {
+        if(mData.driverinput.isSet(DriveTeamInputMap.DRIVER_THROTTLE_AXIS) &&
+           mData.driverinput.isSet(DriveTeamInputMap.DRIVER_TURN_AXIS) &&
+           mData.driverinput.isSet(DriveTeamInputMap.DRIVER_SUB_WARP_AXIS)) {
+            double rotate = mData.driverinput.get(DriveTeamInputMap.DRIVER_TURN_AXIS);
+            double throttle = -mData.driverinput.get(DriveTeamInputMap.DRIVER_THROTTLE_AXIS);
+
+//		    throttle = EInputScale.EXPONENTIAL.map(throttle, 2);
+            rotate = EInputScale.EXPONENTIAL.map(rotate, 2);
+            rotate = Util.limit(rotate, 0.7);
+
+            if (mData.driverinput.get(DriveTeamInputMap.DRIVER_SUB_WARP_AXIS) > DRIVER_SUB_WARP_AXIS_THRESHOLD) {
+                throttle *= SystemSettings.kSnailModePercentThrottleReduction;
+                rotate *= SystemSettings.kSnailModePercentRotateReduction;
             }
 
-            double pipelineNum;
-            if(trackingType != null) {
-                if(mDriverInputCodex.isSet(DriveTeamInputMap.DRIVER_NUDGE_SEEK_LEFT)) {
-                    pipelineNum = trackingType.getPipeline();
-                } else if(mDriverInputCodex.isSet(DriveTeamInputMap.DRIVER_NUDGE_SEEK_RIGHT)) {
-                    pipelineNum = trackingType.getPipeline();
-                }
-            }
+            DriveMessage driveMessage = DriveMessage.fromThrottleAndTurn(throttle, rotate);
+            driveMessage.setNeutralMode(NeutralMode.Brake);
+            driveMessage.setControlMode(ControlMode.PercentOutput);
 
-            // Set limelight pipeline
-            // Add command to the queue
+            driveTrain.setDriveMessage(driveMessage);
         }
-        lastRunCommandQueue = runCommandQueue;
     }
 
-
-    private void updateDriveTrain() {
-        double desiredLeftOutput, desiredRightOutput;
-
-        double rotate = mData.driverinput.get(DriveTeamInputMap.DRIVER_TURN_AXIS);
-        rotate = EInputScale.EXPONENTIAL.map(rotate, 2);
-        double throttle = -mData.driverinput.get(DriveTeamInputMap.DRIVER_THROTTLE_AXIS);
-//		throttle = EInputScale.EXPONENTIAL.map(throttle, 2);
-
-//		if(mElevatorModule.decelerateHeight())
-//		{
-        // throttle = Utils.clamp(throttle, 0.5);
-//		}
-        if (mData.driverinput.get(DriveTeamInputMap.DRIVER_SUB_WARP_AXIS) > DRIVER_SUB_WARP_AXIS_THRESHOLD) {
-            throttle *= SystemSettings.SNAIL_MODE_THROTTLE_LIMITER;
-            rotate *= SystemSettings.SNAIL_MODE_ROTATE_LIMITER;
+    /**
+     * Commands the superstructure to start vision tracking depending on
+     * button presses.
+     */
+    protected void updateVisionCommands() {
+
+        ETrackingType trackingType = null;
+        // Switch the limelight to a pipeline and track
+        if(mDriverInputCodex.isSet(DriveTeamInputMap.DRIVER_TRACK_TARGET_BTN)) {
```

Comment:
Is it possible to have multiple targets selected? If so we need to handle this better.

Reply:
I created a Trello card to handle this scenario.